File: frontend/src/App/types/appConfiguration.ts

export type AppConfiguration = {
    environment?: string,
    environmentDisplayName?: string,
    version?: string,
    keycloakConf: KeycloakConfiguiration,
}

export type KeycloakConfiguiration = {
    authority: string,
    client_id: string,
    redirect_uri: string,
    onSigninCallback: any,
    store: any
}

File: backend/Gateways/Gateway.Web/Configuration/AppConfiguration.cs
namespace Gateway.Web.Configuration;

public class AppConfiguration
{
    public string Environment { get; set; } = default!;
    public string EnvironmentDisplayName { get; set; } = default!;
    public string Version { get; set; } = default!;
    public KeycloakConfiguration Keycloak { get; set; } = default!;
}

public class KeycloakConfiguration
{
    public Uri AuthUrl { get; set; } = default!;
    public string Realm { get; set; } = default!;
    public string ClientId { get; set; } = default!;
}